src/publishers/tradesPublisher.ts

```diff
@@ -0,0 +1,210 @@
+import { logger, setLogLevel } from '../utils/logger';
+import { sleep } from '../utils/utils';
+import { RedisClient } from '../utils/redisClient';
+import { fromEvent, filter, map } from 'rxjs';
+
+require('dotenv').config();
+const driftEnv = (process.env.ENV || 'devnet') as DriftEnv;
+const commitHash = process.env.COMMIT;
+const REDIS_HOST = process.env.REDIS_HOST || 'localhost';
+const REDIS_PORT = process.env.REDIS_PORT || '6379';
+const REDIS_PASSWORD = process.env.REDIS_PASSWORD;
+
+//@ts-ignore
+const sdkConfig = initialize({ env: process.env.ENV });
+
+const stateCommitment: Commitment = 'processed';
+const ORDERBOOK_UPDATE_INTERVAL = 1000;
+
+let driftClient: DriftClient;
+
+const opts = program.opts();
+setLogLevel(opts.debug ? 'debug' : 'info');
+
+const endpoint = process.env.ENDPOINT;
+const wsEndpoint = process.env.WS_ENDPOINT;
+logger.info(`RPC endpoint: ${endpoint}`);
+logger.info(`WS endpoint:  ${wsEndpoint}`);
+logger.info(`DriftEnv:     ${driftEnv}`);
+logger.info(`Commit:       ${commitHash}`);
+
+const main = async () => {
+	const wallet = new Wallet(new Keypair());
+	const clearingHousePublicKey = new PublicKey(sdkConfig.DRIFT_PROGRAM_ID);
+
+	const connection = new Connection(endpoint, {
+		wsEndpoint: wsEndpoint,
+		commitment: stateCommitment,
+	});
+
+	const bulkAccountLoader = new BulkAccountLoader(
+		connection,
+		stateCommitment,
+		ORDERBOOK_UPDATE_INTERVAL
+	);
+
+	driftClient = new DriftClient({
+		connection,
+		wallet,
+		programID: clearingHousePublicKey,
+		accountSubscription: {
+			type: 'polling',
+			accountLoader: bulkAccountLoader,
+		},
+		env: driftEnv,
+		userStats: true,
+	});
+
+	const slotSubscriber = new SlotSubscriber(connection, {});
+
+	const lamportsBalance = await connection.getBalance(wallet.publicKey);
+	logger.info(
+		`DriftClient ProgramId: ${driftClient.program.programId.toBase58()}`
+	);
+	logger.info(`Wallet pubkey: ${wallet.publicKey.toBase58()}`);
+	logger.info(` . SOL balance: ${lamportsBalance / 10 ** 9}`);
+
+	await driftClient.subscribe();
+	driftClient.eventEmitter.on('error', (e) => {
+		logger.info('clearing house error');
+		logger.error(e);
+	});
+
+	await slotSubscriber.subscribe();
+
+	const userMap = new UserMap(
+		driftClient,
+		driftClient.userAccountSubscriptionConfig,
+		false
+	);
+	await userMap.subscribe();
+
+	const redisClient = new RedisClient(REDIS_HOST, REDIS_PORT, REDIS_PASSWORD);
+	await redisClient.connect();
+
+	const eventSubscriber = new EventSubscriber(connection, driftClient.program, {
+		maxTx: 8192,
+		maxEventsPerType: 4096,
+		orderBy: 'client',
+		commitment: 'confirmed',
+		logProviderConfig: {
+			type: 'polling',
+			frequency: 1000,
+		},
+	});
+
+	await eventSubscriber.subscribe();
+
+	const eventObservable = fromEvent(eventSubscriber.eventEmitter, 'newEvent');
+	eventObservable
+		.pipe(
+			filter(
+				(event) =>
+					event.eventType === 'OrderActionRecord' &&
+					JSON.stringify(event.action) === JSON.stringify(OrderAction.FILL)
```

Comment:
i dont think this will work, might need to do this instead:

```suggestion
					isVariant(event.action as OrderAction, 'fill')
```

Reply:
it worked in my testing and isVariant just ends up doing the JSON stringified equivalence under the hood!